scripts/solve_network.py

```diff
@@ -150,9 +150,31 @@ def add_chp_constraints(n):
         define_constraints(n, lhs, "<=", 0, 'chplink', 'backpressure')
 
 
+
+def add_pipe_retrofit_constraint(n):
+    """Add constraint for retrofitting existing CH4 pipelines to H2 pipelines."""
+
+    gas_pipes_i = n.links[n.links.carrier=="Gas pipeline"].index
+    h2_retrofitted_i = n.links[n.links.carrier=='H2 pipeline retrofitted'].index
+
+    if h2_retrofitted_i.empty or gas_pipes_i.empty: return
+
+    link_p_nom = get_var(n, "Link", "p_nom")
+
+    pipe_capacity = n.links.loc[gas_pipes_i, 'p_nom']
+    # according to hydrogen backbone strategy (April, 2020) p.15
+    # https://gasforclimate2050.eu/wp-content/uploads/2020/07/2020_European-Hydrogen-Backbone_Report.pdf
+    # 60% of original natural gas capacity could be used in cost-optimal case as H2 capacity
+    lhs = linexpr((1, link_p_nom.loc[h2_retrofitted_i].rename(index=lambda x: x.replace("H2 pipeline retrofitted", "Gas pipeline"))),
+                  (-0.6, link_p_nom.loc[gas_pipes_i]))
+
+    define_constraints(n, lhs, "=", 0., 'Link', 'pipe_retrofit')
```

Comment:
I am not sure I understand this constraint. It enforces

```
P_H2 = 0.8 * P_CH4
```

where both H2 and CH4 pipelines are expandable. Doesn't that say: if you build H2 retrofitted pipeline also increase gas pipeline capacity?

### Alternative where CH4 network CANNOT be expanded beyond today's capacity

(that might be a reasonable assumption if current CH4 is more than ever would be needed)

Then -- I think -- the constraint should be

```
P_H2 / 0.8 + P_CH4 = P_CH4_initial
```

where the factor 0.8 does the MW_H2 to MW_CH4 conversion.

In this case also the capital cost of the gas pipelines maybe have to be 0. Otherwise, you make money be decreasing capacity.

### Alternative where CH4 network CAN be expanded beyond today's capacity

Then -- I think -- the constraint should be

```
P_H2 / 0.8 + P_CH4 >= P_CH4_initial
```

where the retrofitted pipeline capacity is already limited by the `p_nom_max`:

```
P_H2 <= 0.8 P_CH4_init
```

Then the constraint does not limit the P_CH4.

In this case, however, the capital cost of the gas pipelines must not be 0 (expanding the gas network beyond today's level costs money). But on the other hand, you make money by decreasing capacity. This is a little bit tricky...

### Way out?

The cleanest version might be to go with the first variant and also allow a greenfield gas network expansion (if needed e.g. to connect nodes that are disconnected from gas supply). But more variables...

Reply:
Ah yes, you are right. I would prefer this version `P_H2 / 0.6 + P_CH4 = P_CH4_initial`. We can allow greenfield gas pipes e.g. to isolated nodes, in general the pipe capacities should be sufficient for future gas demand.